File: C++/medium/347_top_k_frequent_elements.cc
#include <iostream>
#include <queue>

using namespace std;

class Solution
{
public:
  vector<int> topKFrequent(vector<int> &nums, int k)
  {
    unordered_map<int, int> freq;
    for (int num : nums)
      freq[num]++;

    vector<vector<int>> buckets(nums.size() + 1);
    for (auto p : freq)
      buckets[p.second].push_back(p.first);

    vector<int> res;
    for (int i = buckets.size() - 1; i >= 0 && res.size() < k; i--)
    {
      for (int j = 0; j < buckets[i].size() && res.size() < k; j++)
        res.push_back(buckets[i][j]);
    }

    return res;
  }
};

File: Python/medium/347_top_k_frequent_elements.py
from typing import List


class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts = {}
        for x in nums:
            counts[x] = counts.get(x, 0) + 1

        return sorted(counts, key=counts.get, reverse=True)[:k]
